### DISCUSSION TITLE
tar_visnetwork with a .Rmd pipeline in data-raw

### DISCUSSION THREAD

**Discussion by TylerGrantSmith** [ropensci/targets#694] [Help]
Currently, I have my targets pipeline defined in an rmarkdown file in a `data-raw` subdirectory.  The pipeline runs and knits fine (creating a `data-raw/_targets.R` script.  How can I run `tar_visnetwork` from the console to view this pipeline?  Right now, it looks like eventually `callr_inner` will `eval` the script, but it is using the project's root directory.  
```
X = list.files(
    path = file.path("_targets_r", "globals")
...
```

If I manually `setwd("data-raw")` then it will run fine, but how can I do this without that manual intervention?

Thanks for answering all my questions.

**Comment by wlandau** [comment]
`tar_visnetwork(script = "data-raw/_targets.R", store = "data-raw/_targets")` will probably work. To configure the default `script` and `store`, you can use `tar_config_set()`, which writes to a `_targets.yaml` file in your working directory. https://books.ropensci.org/targets/projects.html has details about managing multiple projects and managing projects in non-standard locations.

**Comment by TylerGrantSmith** [comment]
Unfortunately, I tried that already and it doesn't work.  It is because the `data-raw/_targets.R` script that `tar_make` creates when the markdown is knit uses path names relative to `data-raw/_targets.R`.  It is assuming that `_targets.R` is in the working directory, which is TRUE when the file is knitting, but not true when you want to work with the pipeline interactively.

```yaml
main:
  script: data-raw/_targets.R
  store: data-raw/_targets/
  reporter_make: summary
```

Running `tar_make` from the project root with this configuration does nothing because it will execute the auto-generated `data-raw_targers.R` and the file paths are incorrect.  You can see that it will list files from `/_targets_r/globals` rather than `/data-raw/targets_r/globals`:

```r
# Generated by targets: do not edit by hand
library(targets)
lapply(
  X = list.files(
    path = file.path("_targets_r", "globals"),
    pattern = "\\.R$",
    full.names = TRUE,
    all.files = TRUE,
    recursive = TRUE
  ),
  FUN = function(path) {
    eval(
      expr = parse(text = readLines(con = path, warn = FALSE)),
      envir = targets::tar_option_get(name = "envir")
    )
  }
)
lapply(
  X = list.files(
    path = file.path("_targets_r", "targets"),
    pattern = "\\.R$",
    full.names = TRUE,
    all.files = TRUE,
    recursive = TRUE
  ),
  FUN = function(path) {
    eval(
      expr = parse(text = readLines(con = path, warn = FALSE)),
      envir = targets::tar_option_get(name = "envir")
    )
  }
)
```

**Reply by wlandau** [reply]
So then you will probably have better luck with `withr::with_dir("data-raw", tar_visnetwork())`. If you want to inspect the pipeline outside `data-raw`, maybe you could run the pipeline from there as well. You could run the report from the parent directory of `data-raw` and use `tar_config_set()` to establish the script and store inside `data-raw` instead of running the report from inside `data-raw`.
